src/tools/return_results/implementation.js

```diff
@@ -0,0 +1,107 @@
+import { BaseTool } from '../common/base-tool.js';
+
+class ReturnResultsTool extends BaseTool {
+    constructor() {
+        super('return_results', 'Signals task completion and returns structured results');
+
+        this.requiredParams = ['result'];
+        this.parameterTypes = {
+            result: 'object',
+        };
+    }
+
+    async implementation(params) {
+        const { result } = params;
+
+        try {
+            // Validate result structure
+            this._validateResultStructure(result);
+
+            const agentManager = this.context.agentManager;
+            if (!agentManager) {
+                throw new Error('AgentManager not available in context');
+            }
+
+            // Get current agent ID from context or determine from role
+            // Results are sent as messages to the parent supervisor
+            const currentAgentId = this.context?.currentAgentId || this.context?.agentId;
+
+            if (!currentAgentId) {
+                // If no agent ID in context, this might be called by the main supervisor
+                // In that case, we need to handle it differently
+                throw new Error(
+                    'Cannot determine current agent ID for result reporting. This tool should only be called by worker agents.'
+                );
+            }
+
+            // Add completion metadata
+            const enrichedResult = {
+                ...result,
+                completed_at: new Date().toISOString(),
+                agent_id: currentAgentId,
+            };
+
+            // Report result to AgentManager
+            await agentManager.reportResult(currentAgentId, enrichedResult);
+
+            return this.createSuccessResponse({
+                task_completed: true,
+                agent_id: currentAgentId,
+                result_status: result.status,
+                summary: result.summary,
```

Comment:
response should not contain back information agent has just send it should contain message
"results has been send to... wait for response or further instructions"